File: problems/795_number_of_subarrays_with_bounded_maximum/Solution.java
class Solution {
    public int numSubarrayBoundedMax(int[] nums, int left, int right) {
        return count(nums, right) - count(nums, left - 1);
    }

    private int count(int[] nums, int bound) {
        int ans = 0, cnt = 0;
        for (int num : nums) {
            cnt = num <= bound ? cnt + 1 : 0;
            ans += cnt;
        }
        return ans;
    }
}

File: problems/795_number_of_subarrays_with_bounded_maximum/solution.py
from typing import List


class Solution:
    def numSubarrayBoundedMax(self, nums: List[int], left: int, right: int) -> int:
        def count(bound):
            ans = cnt = 0
            for num in nums:
                cnt = cnt + 1 if num <= bound else 0
                ans += cnt
            return ans
        return count(right) - count(left - 1)
